src/main/groovy/de/felixschulze/gradle/HockeyAppUploadTask.groovy

```diff
@@ -75,14 +75,17 @@ class HockeyAppUploadTask extends DefaultTask {
 
         hockeyApp = project.hockeyapp
 
-        // Get the first output apk file if android
+        // Get all output apk files if android
         if (applicationVariant) {
             logger.debug('Using android application variants')
 
             applicationVariant.outputs.each {
                 if (FilenameUtils.isExtension(it.outputFile.getName(), "apk")) {
-                    applicationFile = it.outputFile
-                    return true
+                    if (it.outputFile.exists()) {
```

Comment:
If I'm not mistaken, test APKs listed in TestVariant, not in ApplicationVariant. At the meantime TestVariant has getTestedVariant() method for reference to the corresponding ApplicationVariant. Therefore TestVariant has access to app APKs and test APKs, but ApplicationVariant has only access to app APKs. But I'm not really familiar with all this build system, so I could be wrong.